W-18556711 access based policy gr (#864)
mulesoft/docs-gateway: gateway/1.9/modules/ROOT/pages/policies-included-mcp-attribute-access-control.adoc

>s|Summary      | Controls access to tools, resources, and prompts based on user information such as Tiers, IP, Headers, or Claims
>s|Category | MCP
>s|First Flex Gateway version available | v1.9.3
>s| Returned Status Codes | 400 - Invalid token
|===

== Summary

The MCP Attribute-Based Access Control policy controls access to server tools, resources, and prompts based on user information such as Tiers, IP, Headers, or Claims. Use the https://docs.cedarpolicy.com/[Cedar policy language] to define access rules.

The MCP Attribute-Based Access Control policy requires that another authentication policy be applied before it. The authentication policy can be one of the following included policies or a custom policy:

xref:policies-included-basic-auth-ldap.adoc[]
xref:policies-included-basic-auth-simple.adoc[]
xref:policies-included-client-id-enforcement.adoc[]
xref:policies-included-jwt-validation.adoc[]
xref:policies-included-oauth-token-introspection.adoc[]
xref:policies-included-openid-token-enforcement.adoc[]
xref:policies-included-rate-limiting-sla.adoc[]

> For custom authentication policies, the custom policy must authenticate by using the `Authentication` injectable. To learn more, see pdk::policies-pdk-configure-features-authentication.adoc[]

Reviewer: ```suggestion
For custom authentication policies, the custom policy must authenticate by using the `Authentication` injectable. To learn more, see pdk::policies-pdk-configure-features-authentication.adoc[].
```

== Configuring Policy Parameters

include::partial$policy-title-headers.adoc[tag=configFileTitleOnly]

The MCP Attribute-Based Access Control policy is not supported in Local Mode.

include::partial$policy-title-headers.adoc[tag=ui]

[%header%autowidth.spread,cols="a,a"]

|===
| Element | Description

> | Rules | Rules to grant requests access to tools, resources, and prompts written in the https://docs.cedarpolicy.com/[Cedar policy language].

Reviewer: ```suggestion
| Rules | Rules to control access to tools, resources, and prompts, written in the https://docs.cedarpolicy.com/[Cedar policy language].
```

| Cedar Expression | Cedar expression defining the access rule. For example: `permit(principal,action == Action::"tools/call` or `resource == Tool::"calculator");`

> | Authentication Type | Type of authentication used by the authentication policy applied before this policy.

Reviewer: ```suggestion
| Authentication Type | Authentication type specified by the authentication policy, which must be applied before this policy.
```
